### PR TITLE
Rename and refactor to use `RzBitVector` instead of `RzILBitVector`

### LINKED ISSUE #2010: Move RzILBitVector to RzUtil
The bitvector is very generic and useful in many places.
It should be renamed from `RzILBitVector` to just `RzBitVector` and moved to RzUtil.

### PR DESCRIPTION
**Pull request by DMaroo**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactor and extract `RzILBitVector` from `rz_il` to `rz_util` for more generalized use.

**Test plan**

Green CI.

**Closing issues**

Closes #2010.

### CODE REVIEW

File: test/unit/test_il_definitions.c [hunk 1]
```diff
@@ -17,483 +17,483 @@ bool test_rzil_bv_init32(void) {
...
-	rz_il_bv_free(bits);
-	rz_il_bv_free(bits_cmp);
-	rz_il_bv_free(bits_dup);
+	rz_bitvector_free(bits);
+	rz_bitvector_free(bits_cmp);
+	rz_bitvector_free(bits_dup);
 	mu_end;
 }
 
 bool test_rzil_bv_init64(void) {
```
Comment (XVilka) [inline test/unit/test_il_definitions.c]: It makes sense to separate just bitvector unit tests from the `test_il_definitions.c` into a new C file, with correspondingly named functions.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by thestr4ng3r** [comment]
I think we can keep the `bv` naming, i.e. have the `RzBitVector` type and `rz_bv_*` functions. I expect bitvectors to be used in many places of the code and `bv` is way more convenient. But I would also like to hear other opinions @XVilka @ret2libc @wargio.

**Comment by ret2libc** [comment]
> I think we can keep the `bv` naming, i.e. have the `RzBitVector` type and `rz_bv_*` functions. I expect bitvectors to be used in many places of the code and `bv` is way more convenient. But I would also like to hear other opinions @XVilka @ret2libc @wargio.

I'm ok with `rz_bv_*`.

**Comment by wargio** [comment]
> more convenient. But I would also like to hear other o

same. i'm ok with that naming convention.

**Comment by XVilka** [comment]
`rz_bv_` is ok for me. But I think it's time to move to the RzUtil, why not?

**Review by XVilka — APPROVED** [review]

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**